### ISSUE TITLE
Tracking rebase branches

### ISSUE THREAD

**Issue by lws-team** [lvc/abi-monitor#3]
Hi -

Thanks for your cool tool.

I have it set up on a cronjob to output daily updates for my project here

https://libwebsockets.org/abi/timeline/libwebsockets/index.html

However, on master there is no attempt to keep a history in git.  If there are problems in patches from the last day or two, or sometimes further back, the patches are simply rewritten with the fix.  The stable branches are treated as "patches on top" -only history branches.

This makes trouble for abi-monitor, because he wants to use git pull.  git pull will reject the fast forward normally, if the update breaks history so you "lose" content in your current HEAD.

I can work around it by editing abi-monitor script to use "git fetch" instead.  But it'd perhaps be better if there was some way to select this in the config.

Thanks.

**Comment by lvc** [comment]
Great news!

As I understand the `git fetch` command doesn't update files in the source tree, so it cannot replace the `git pull` because the tracker will build and analyze old versions of files. May be for this reason the git log https://libwebsockets.org/abi/changelog/libwebsockets/current/log.html is not updated too since 2016-04-18. Am I right?

Thank you.

**Comment by lws-team** [comment]
I guess you are right about something there, but I get a cronjob transcript like this

```
Searching for new packages
No new packages found
Updating source code in repository
From https://github.com/warmcat/libwebsockets
 + 9316076...cc704c7 master     -> origin/master  (forced update)
Building 'current'
Detecting date of current
Detecting soname of current
Creating changelog for current
Creating ABI dump for current
Creating ABI dump for lib/libwebsockets.so.7
Creating objects ABI report between 1.7.5 and current
Creating ABICC report for lib/libwebsockets.so.7 (1.7.5) and lib/libwebsockets.so.7 (current)
Diff headers 1.7.5 and current
The index has been generated to: timeline/libwebsockets/index.html
The index has been generated to: timeline/libwebsockets/index.html
Deploy to /var/www/libwebsockets.org/abi
Copy timeline/libwebsockets
Copy headers_diff/libwebsockets
Copy changelog/libwebsockets
Copy abi_dump/libwebsockets
Copy objects_report/libwebsockets
Copy compat_report/libwebsockets
Copy db/libwebsockets
Copy css
```

This one is from the 20th... it's not impossible something else is wrong with my cronjob arrangements but the git part seems to go well.

**Comment by lws-team** [comment]
Hm no it seems to have been recooked well at the time of the cronjob and the report is generated and it's copied into place correctly, currently it's telling `Last updated on Fri Apr 22 08:22:30 2016`

Where does that date in the "Date" field actually come from?  Although the last few days were spent getting one patch right by amending it over and over, actually for a few days the patch dates for the prior patches have been the 20th.  It looks like that has never been updated since the first time I ran it by hand.

**Comment by lws-team** [comment]
Yes the problem is it went into the local repo's "origin/master" and what's checked out was "master".

Another hack on your script like this

```
    if(-d $CurRepo)
    {
        chdir($CurRepo);

        if($Git)
        {
            printMsg("INFO", "Updating source code in repository");
            my $Log = qx/git fetch/;

            if($Log=~/Already up\-to\-date/i) {
                $UpToDate = 1;
            }

            my $Log = qx/git reset --hard origin\/master/;
        }
```

gets it working but that's probably not what a nice solution looks like...

**Comment by vittalll** [comment]
> Hi -
> 
> Thanks for your cool tool.
> 
> I have it set up on a cronjob to output daily updates for my project here
> 
> https://libwebsockets.org/abi/timeline/libwebsockets/index.html
> 
> However, on master there is no attempt to keep a history in git. If there are problems in patches from the last day or two, or sometimes further back, the patches are simply rewritten with the fix. The stable branches are treated as "patches on top" -only history branches.
> 
> This makes trouble for abi-monitor, because he wants to use git pull. git pull will reject the fast forward normally, if the update breaks history so you "lose" content in your current HEAD.
> 
> I can work around it by editing abi-monitor script to use "git fetch" instead. But it'd perhaps be better if there was some way to select this in the config.
> 
> Thanks.



> Hi -
> 
> Thanks for your cool tool.
> 
> I have it set up on a cronjob to output daily updates for my project here
> 
> https://libwebsockets.org/abi/timeline/libwebsockets/index.html
> 
> However, on master there is no attempt to keep a history in git. If there are problems in patches from the last day or two, or sometimes further back, the patches are simply rewritten with the fix. The stable branches are treated as "patches on top" -only history branches.
> 
> This makes trouble for abi-monitor, because he wants to use git pull. git pull will reject the fast forward normally, if the update breaks history so you "lose" content in your current HEAD.
> 
> I can work around it by editing abi-monitor script to use "git fetch" instead. But it'd perhaps be better if there was some way to select this in the config.

Hi sir,

i am using abi-monitor tool bellow command is download particular kernel
abi-monitor -get -v 5.15.38 linux.json
this above command is working fine but after the download i am trying to build that kernel version that's why i am passing bellow
command
abi-monitor -build -v 5.15.38 linux.json
but its not build properly its showing error in bellow

ERROR : custom build has failed (exit code 256)
ERROR : failed to build
any suggestion
